### DISCUSSION TITLE
Code server remote Docker.sock connection

### DISCUSSION THREAD

**Discussion by bigAboo80** [coder/code-server#4825] [Q&A]
Hi, I bind volume in my code server container to var/run/docker.sock and add user to docker group , after that error about “Is docker installed “ changed to “error connect: EACCES/var/run/docker.sock
we to dig ?)

**Comment by jsjoeio** [comment]
Hmmm...Can you explain how/where you're running this? Is this a local machine or a remote VM? 

And maybe steps to reproduce? Otherwise, we have docs [here](https://coder.com/docs/code-server/latest/install#docker) about running code-server with Docker

**Reply by bigAboo80** [reply]
- Hi, i have a bare metal server, at home, and set up on it docker, portainer and code server in container and bind volume mount to container, /var/run/docker.dock:/var/run/docker.sock
When i open code-server, I installed there docker extension, and ssh fs, to work with docker,. 
This post seems like about same issue?

[https://github.com/coder/code-server/discussions/1848]()

But I can't install docker-ce-cli in container 
Get error broken pipe.
So i want to code-server container can manage with docker itself on ubuntu server or on others ssh connected servers.

**Reply by jsjoeio** [reply]
> Get error broken pipe.

Can you post the error logs?

**Comment by bigAboo80** [comment]
```
root@f9be147f2c61:/# apt install docker-ce-cli
Reading package lists... Done
Building dependency tree       
Reading state information... Done
The following NEW packages will be installed:
  docker-ce-cli
0 upgraded, 1 newly installed, 0 to remove and 0 not upgraded.
33 not fully installed or removed.
Need to get 40,7 MB of archives.
After this operation, 164 MB of additional disk space will be used.
Get:1 https://download.docker.com/linux/ubuntu focal/stable amd64 docker-ce-cli amd64 5:20.10.12~3-0~ubuntu-focal [40,7 MB]
Fetched 40,7 MB in 2s (17,0 MB/s)           
debconf: unable to initialize frontend: Dialog
debconf: (No usable dialog-like program is installed, so the dialog based frontend cannot be used. at /usr/share/perl5/Debconf/FrontEnd/Dialog.pm line 76, <> line 1.)
debconf: falling back to frontend: Readline
(Reading database ... 18614 files and directories currently installed.)
Preparing to unpack .../docker-ce-cli_5%3a20.10.12~3-0~ubuntu-focal_amd64.deb ...
Unpacking docker-ce-cli (5:20.10.12~3-0~ubuntu-focal) ...
dpkg: error processing archive /var/cache/apt/archives/docker-ce-cli_5%3a20.10.12~3-0~ubuntu-focal_amd64.deb (--unpack):
 unable to make backup link of './usr/bin/docker' before installing new version: Invalid cross-device link
dpkg-deb: error: paste subprocess was killed by signal (Broken pipe)
Errors were encountered while processing:
 /var/cache/apt/archives/docker-ce-cli_5%3a20.10.12~3-0~ubuntu-focal_amd64.deb
E: Sub-process /usr/bin/dpkg returned an error code (1)
root@f9be147f2c61:/
```

**Reply by lg-nineoneninethreenine** [reply]
This looks like a problem with APT, can you run this in root mode?
```
# dpkg --configure -a
```

**Reply by bigAboo80** [reply]
after dpkg --configure -a
`dpkg --configure -a
Setting up systemd-sysv (245.4-4ubuntu3.15) ...
Setting up xdg-user-dirs (0.17-2ubuntu1) ...
Setting up libip6tc2:amd64 (1.8.4-3ubuntu2) ...
Setting up libglib2.0-0:amd64 (2.64.6-1~ubuntu20.04.4) ...
No schema files found: doing nothing.
Setting up libicu66:amd64 (66.1-2ubuntu2.1) ...
Setting up docker-scan-plugin (0.12.0~ubuntu-focal) ...
Setting up libnss-systemd:amd64 (245.4-4ubuntu3.15) ...
First installation detected...
Checking NSS setup...
Setting up apparmor (2.13.3-7ubuntu5.1) ...
debconf: unable to initialize frontend: Dialog
debconf: (No usable dialog-like program is installed, so the dialog based frontend cannot be used. at /usr/share/perl5/Debconf/FrontEnd/Dialog.pm line 76.)
debconf: falling back to frontend: Readline
Created symlink /etc/systemd/system/sysinit.target.wants/apparmor.service → /lib/systemd/system/apparmor.service.
Setting up libglib2.0-data (2.64.6-1~ubuntu20.04.4) ...
dpkg: dependency problems prevent configuration of docker-ce:
 docker-ce depends on docker-ce-cli; however:
  Package docker-ce-cli is not installed.

dpkg: error processing package docker-ce (--configure):
 dependency problems - leaving unconfigured
Setting up libdbus-1-3:amd64 (1.12.16-2ubuntu2.1) ...
Setting up dbus (1.12.16-2ubuntu2.1) ...
Setting up libmnl0:amd64 (1.0.4-2) ...
Setting up containerd.io (1.4.12-1) ...
Created symlink /etc/systemd/system/multi-user.target.wants/containerd.service → /lib/systemd/system/containerd.service.
Setting up libxtables12:amd64 (1.8.4-3ubuntu2) ...
Setting up libltdl7:amd64 (2.4.6-14) ...
Setting up dmsetup (2:1.02.167-1ubuntu1) ...
Setting up pigz (2.4-1) ...
Setting up libnfnetlink0:amd64 (1.0.1-3build1) ...
Setting up libpam-systemd:amd64 (245.4-4ubuntu3.15) ...
debconf: unable to initialize frontend: Dialog
debconf: (No usable dialog-like program is installed, so the dialog based frontend cannot be used. at /usr/share/perl5/Debconf/FrontEnd/Dialog.pm line 76.)
debconf: falling back to frontend: Readline
Setting up libgirepository-1.0-1:amd64 (1.64.1-1~ubuntu20.04.1) ...
Setting up libxml2:amd64 (2.9.10+dfsg-5ubuntu0.20.04.1) ...
Setting up python3-dbus (1.2.16-1build1) ...
Setting up slirp4netns (0.4.3-1) ...
Setting up libnftnl11:amd64 (1.1.5-1) ...
Setting up dbus-user-session (1.12.16-2ubuntu2.1) ...
Setting up shared-mime-info (1.15-1) ...
Setting up gir1.2-glib-2.0:amd64 (1.64.1-1~ubuntu20.04.1) ...
Setting up libnetfilter-conntrack3:amd64 (1.0.7-2) ...
Setting up docker-ce-rootless-extras (5:20.10.12~3-0~ubuntu-focal) ...
Setting up iptables (1.8.4-3ubuntu2) ...
update-alternatives: using /usr/sbin/iptables-legacy to provide /usr/sbin/iptables (iptables) in auto mode
update-alternatives: using /usr/sbin/ip6tables-legacy to provide /usr/sbin/ip6tables (ip6tables) in auto mode
update-alternatives: using /usr/sbin/arptables-nft to provide /usr/sbin/arptables (arptables) in auto mode
update-alternatives: using /usr/sbin/ebtables-nft to provide /usr/sbin/ebtables (ebtables) in auto mode
Setting up python3-gi (3.36.0-1) ...
Setting up networkd-dispatcher (2.1-2~ubuntu20.04.1) ...
Created symlink /etc/systemd/system/multi-user.target.wants/networkd-dispatcher.service → /lib/systemd/system/networkd-dispatcher.service.
Processing triggers for systemd (245.4-4ubuntu3.15) ...
Processing triggers for libc-bin (2.31-0ubuntu9.2) ...
Errors were encountered while processing:
 docker-ce`

**Reply by bigAboo80** [reply]
apt --fix-broken install points me to unfinished docker-ce-cli  installation an it come around and around...

**Reply by lg-nineoneninethreenine** [reply]
Cna you try getting the DEB file of the app?

**Reply by bigAboo80** [reply]
` dpkg -i docker-ce-cli_20.10.12~3-0~ubuntu-focal_amd64.deb 
(Reading database ... 16745 files and directories currently installed.)
Preparing to unpack docker-ce-cli_20.10.12~3-0~ubuntu-focal_amd64.deb ...
Unpacking docker-ce-cli (5:20.10.12~3-0~ubuntu-focal) ...
dpkg: error processing archive docker-ce-cli_20.10.12~3-0~ubuntu-focal_amd64.deb (--install):
 unable to make backup link of './usr/bin/docker' before installing new version: Invalid cross-device link
dpkg-deb: error: paste subprocess was killed by signal (Broken pipe)
Errors were encountered while processing:
 docker-ce-cli_20.10.12~3-0~ubuntu-focal_amd64.deb`

**Reply by lg-nineoneninethreenine** [reply]
This seems to me like a bug on Docker's side. Have you raised a ticket or does the same app work anywhere else?

**Reply by bigAboo80** [reply]
No
